### ISSUE TITLE
Coreos user-data file uses public ip for fleet, firewall script brings fleet down

### ISSUE THREAD

**Issue by geeksoul-me** [deis/deis#4594]
Tested on Deis 1.11.

User-data example file in contrib/coreos/user-data.example is using public ip for fleet. As etcd works with private ip, applying custom_firewall.sh on digital ocean causes fleet to fail with etcd errors.

I was able to work around this, by replacing
```
fleet:
    public-ip: $public_ipv4
```
with 
```
fleet:
    public-ip: $private_ipv4
```
in contrib/coreos/user-data file , after running ```make discovery-url```

**Comment by yebyen** [comment]
+1 I was literally just having this issue myself and saw your issue pop up on #deis irc.  I will try out your workaround!  Myself I noticed that jq does not appear to be able to write itself to /opt/bin when the custom-firewall script gets executed, not sure if that's related Thanks!

**Comment by yebyen** [comment]
Well, looks like jq lives in /usr/bin/ now, and doesn't need to be downloaded by contrib script so that curl error was a red herring: your issue was exactly my issue.  So that worked for both of us.  Thanks again!

**Comment by geeksoul-me** [comment]
Cheers :)

**Comment by geeksoul-me** [comment]
So, I think this is a bug. Not sure though.

**Comment by yebyen** [comment]
I'm sure it's a bug, I'm just not sure it's the same bug on all platforms.  I for one only use DO so can't say if there's a reason to keep it how it is.

**Comment by carmstrong** [comment]
I suspect the root cause is the platform-specific service which infers a host's public/private addresses.

We recently discovered a similar issue on Azure: https://github.com/coreos/bugs/issues/904

**Comment by jgmize** [comment]
I needed to apply this workaround today when building a couple of 1.11.1 clusters in AWS.

**Comment by sstarcher** [comment]
We have the same issue in AWS.  This is a breaking change for us in AWS, but easy to fix. 
The commit was 445dbaecfadb1e2690734413cc43f30abcb13b21

**Comment by mboersma** [comment]
This will be fixed in upcoming Deis v1.11.2 for new clusters, but for existing v1.11.0 or v1.11.1 clusters, it will be necessary to visit each host, read the contents of /etc/environment, edit the fleet.service to contain `FLEET_PUBLIC_IPV4=[COREOS_PRIVATE_IPV4_value]`, and `systemctl restart fleet`.

A thousand apologies for this mistake and its painful workaround. :person_frowning:

**Comment by nathansamson** [comment]
> but for existing v1.11.0 or v1.11.1 clusters, it will be necessary to visit each host

So theoratically. What would happen if I do this on 3 out of 5 nodes. And then suddenly something goes wrong, and I forget to do the remaining 2 nodes? Will they still be able to communicate with each ather (although some over private network, some over public network) or will all hell break loose?

Except for a non working firewall script, and probably slightly worse reaction speed on new fleet changes, what are the drawbacks of keeping it as is?
